src/collective/ifttt/actions/ifttt.py

```diff
@@ -0,0 +1,178 @@
+import requests
+
+
+logger = logging.getLogger('collective.ifttt')
+
+PAYLOAD_DESCRIPTION = 'description'
+PAYLOAD_USERNAME = 'username'
+PAYLOAD_START = 'start'
+
+payload_options = SimpleVocabulary([
+    SimpleTerm(value=PAYLOAD_DESCRIPTION, title=_(u'Description of action')),
+    SimpleTerm(value=PAYLOAD_USERNAME, title=_(u'Username of Editor')),
+    SimpleTerm(value=PAYLOAD_START, title=_(u'Event Start Date/Time'))
+])
+
+
+class IIftttTriggerAction(Interface):
+    """
+        Definition of the configuration available for a  Ifttt action
+    """
+    ifttt_event_name = schema.TextLine(
+        title=_(u'Ifttt applet name'),
+        description=_(
+            u'Give the name of Ifttt applet which you want to trigger'
+        ),
+        required=True,
+    )
+
+    payload_option = schema.Choice(
+        title=_(u'Choose 3rd Payload'),
+        vocabulary=payload_options,
+        description=_(
+            u'Choose whether the 3rd payload is Description of Content rule, '
+            u'Username of editor, or Event Start Date/time'
+        ),
+        required=True,
+    )
+
+
+@implementer(IIftttTriggerAction, IRuleElementData)
+class IftttTriggerAction(SimpleItem):
+    """
+        The implementation of the action defined before
+    """
+
+    ifttt_event_name = u''
+    payload_option = u''
+
+    element = 'plone.actions.Ifttt'
+
+    @property
+    def summary(self):
+        return _(
+            u'Trigger IFTTT action ${ifttt_event_name} with context title,'
+            u'url and ${payload_option}',
+            mapping=dict(
+                ifttt_event_name=self.ifttt_event_name,
+                payload_option=self.payload_option,
+            ),
+        )
+
+
+@implementer(IExecutable)
+@adapter(Interface, IIftttTriggerAction, Interface)
+class IftttTriggerActionExecutor(object):
+    """
+        The executor for this action
+    """
+    timeout = 120
+
+    # element is defined with Ifttt Trigger form
+    # context is defined as site content for which content rule is triggered
+    def __init__(self, context, element, event):
+        self.element = element
+        self.context = context
+        self.event = event
+
+    def __call__(self, *args, **kwargs):
+        ifttt_event_name = self.element.ifttt_event_name
+        payload_option = self.element.payload_option
+        title = self.context.Title().decode('utf-8', 'ignore')
+        url = self.context.absolute_url()
+        secret_key = api.portal.get_registry_record('ifttt.ifttt_secret_key')
+        ifttt_trigger_url = 'https://maker.ifttt.com/trigger/' + \
+                            ifttt_event_name + '/with/key/' + secret_key
+        payload = {'title': title, 'url': url}
+        if payload_option == PAYLOAD_DESCRIPTION:
+            payload[payload_option] = self.context.description
+        elif payload_option == PAYLOAD_USERNAME:
+            payload[payload_option] = api.user.get_current()
+        elif payload_option == PAYLOAD_START:
```

Comment:
@Shriyanshagro Let's do this properly.

As I already commented, fallback to effective date would cause confusion and was a bad idea.

We should return start date of an event or nothing.

At first, we should import an interface

```python
from plone.event.interfaces import IEventAccessor
```

This is a new contract in Plone 5 to allow any content that wants to behave like an event to define its start date & time, also decoupling it from the actual storage of the data. This also works for recurring events, because their start date & time should dynamically adapt to the next occurrence.

Then we use the accessor

```python
try:
    payload[payload_option] = IEventAccessor(self.context).start
except TypeError:
    payload[payload_option] = None
```

We have exception for TypeError which happens when the context does implement or have registered adapter for IEventAccessor interface/contract.